### PR TITLE
KAFKA-15662: Add support for clientInstanceIds in Kafka Stream

### PR DESCRIPTION
**Pull request by mjsax**
Part of KIP-714.

Add support to collect client instance id of the global consumer.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java [hunk 1]
```diff
@@ -65,6 +69,9 @@ public class GlobalStreamThread extends Thread {
     private final AtomicLong cacheSize;
     private volatile StreamsException startupException;
     private java.util.function.Consumer<Throwable> streamsUncaughtExceptionHandler;
+    private volatile Uuid globalConsumerClientInstanceId = null;
+    private volatile long fetchDeadline = -1;
```
Comment (lucasbru) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java]: Same, give this a name that is related to clientInstanceId

**Review by lucasbru — APPROVED** [review]
LGTM, but I have the same questions as in https://github.com/apache/kafka/pull/14922

**Review by wcarlson5 — COMMENTED** [review]
lGTM

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java [hunk 2]
```diff
@@ -454,4 +485,19 @@ public void shutdown() {
     public Map<MetricName, Metric> consumerMetrics() {
         return Collections.unmodifiableMap(globalConsumer.metrics());
     }
+
+    public KafkaFuture<Uuid> globalConsumerInstanceId(final Duration timeout) {
```
Comment (wcarlson5) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java:491]: Same comment here if you don't mind

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java [hunk 3]
```diff
@@ -1477,6 +1528,27 @@ public Object getStateLock() {
         return stateLock;
     }
 
+    public Map<String, KafkaFuture<Uuid>> consumerClientInstanceIds(final Duration timeout) {
```
Comment (wcarlson5) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java]: and here

**Review by wcarlson5 — APPROVED** [review]
LGTM no further concerns except the same ones on the other PR

**Review by lucasbru — APPROVED** [review]
LGTM

**Pull request merged**